src/App.js

```diff
@@ -1,16 +1,92 @@
-import React from 'react';
+import React from "react";
+import TodoForm from "./components/TodoComponents/TodoForm";
+import TodoList from "./components/TodoComponents/TodoList";
+
+import "./components/TodoComponents/Todo.css";
+
+const todos = [
+  {
+    task: "Learn React",
+    id: 101,
+    completed: false
+  },
+  {
+    task: "Learn Class Components",
+    id: 102,
+    completed: false
+  },
+  {
+    task: "Create React Todo List",
+    id: 103,
+    completed: false
+  }
+];
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
+  constructor(){
+    super();
+    this.state = {
+      todos //todos: todos
+    };
+  }
+
+  addTask = (e, taskName) => {
+    e.preventDefault();
+    const existing = this.state.todos.filter(
+      task => task.task === taskName
+    );
+    if(existing.length === 0){
+      const newTask = {
+        task: taskName,
+        id: Date.now(),
```

Comment:
I'm glad I figured this out, nice solution to prevent duplicates! maybe add an alert to let the user know what happened as well.